src/hacspec.rs

```diff
@@ -65,486 +43,490 @@ pub fn hex_string_to_bytes(s: &str) -> Vec<u8> {
 }
 
 /// Common trait for all byte arrays.
-pub trait ByteArray {
-    fn raw<'a>(&'a self) -> &'a [u8];
+pub trait SeqTrait<T: Copy> {
+    fn raw<'a>(&'a self) -> &'a [T];
     fn len(&self) -> usize;
-    fn iter(&self) -> std::slice::Iter<u8>;
+    fn iter(&self) -> std::slice::Iter<T>;
 }
 
 // ======================== Variable length arrays ========================== //
 
-#[derive(Copy, Clone, Debug)]
-pub struct ByteSlice<'a> {
-    b: &'a[u8]
-}
-impl<'a> ByteSlice<'a> {
-    fn new(b_in: &'a Bytes) -> ByteSlice<'a> {
-        Self {
-            b: &b_in[..]
-        }
-    }
-}
-
 /// Variable length byte arrays.
 #[derive(Debug, Clone, PartialEq, Default)]
-pub struct Bytes {
-    b: Vec<u8>,
+pub struct Seq<T: Copy> {
+    b: Vec<T>,
 }
 
-impl Bytes {
-    pub fn get_slice<'a>(&'a self) -> ByteSlice<'a> {
-        ByteSlice::new(&self)
-    }
+pub type Bytes = Seq<U8>;
```

Comment:
Namings here might be a little confusing.

Reply:
I can rename to `ByteSeq`. I prefer not to use "slice" since it is already bound to a precise Rust type, which might confuse Rust user.

Reply:
Fair enough. `ByteSeq` would work for me.